src/docfx/build/page/BuildPage.cs

```diff
@@ -37,6 +36,7 @@ public static async Task<List<Error>> Build(Context context, Document file)
 
                 var isConceptual = string.IsNullOrEmpty(file.Mime) || TemplateEngine.IsLandingData(file.Mime);
                 (output, metadata) = ApplyPageTemplate(context, file, mergedMetadata, mergeModel, isConceptual);
+                metadata = metadata is null ? null : new JObject(metadata.Properties().OrderBy(p => p.Name));
```

Comment:
if sort is done here, only `.publish.json` are sorted, `.raw.page.json` and `.mta.json` are not sorted, should sort `mergedMetadata`

Reply:
https://github.com/dotnet/docfx/blob/75c0979bc96d94ab170f6b8a2fccda0fff7ff121/src/docfx/build/page/BuildPage.cs#L78
This could cover `.mta.json` as well, but where do we write `.raw.page.json`

Reply:
This is where content in `.raw.page.json` is created: https://github.com/dotnet/docfx/blob/ba2c1d924c6f1ccb7e1efda2258506b03837e777/src/docfx/build/page/BuildPage.cs#L286

Reply:
ok, then `output` in line 38 would be the content written into `.raw.page.json`. Shall we sort `output` as well? If we only sort `mergedMetadata`, there are other parts merged afterward, they will not be in order, right?

Reply:
But since `pageModel` is actually a strong model, do we care that much about it, do we only focus on the `JObject` part here?

Reply:
there is no merge afterward today, we can sort output when there is need

Reply:
https://github.com/dotnet/docfx/blob/ba2c1d924c6f1ccb7e1efda2258506b03837e777/src/docfx/build/page/BuildPage.cs#L35
`pageModel` is merged into `mergeModel` after `mergedMetadata`, and then `mergedMetadata` and `mergeModel` input into `ApplyPageTemplate` to get `metadata` as content of `.mta.json` and `ExtensionData` in `PublishItem`. I still believe it is more straightforward to just sort `metadata` before written.

Reply:
sure, that make sense in the current code flow. Just ensure both `.raw.page.json` and `.mta.json` are sorted.